Automattic/wp-calypso
# Me: Add an 'I am a developer' toggle

Fixes https://github.com/Automattic/dotcom-forge/issues/2333

## Proposed Changes

Adds an 'I am a developer' toggle to `/me`.

<img width="787" alt="image" src="https://user-images.githubusercontent.com/36432/236019432-9b788cde-e3db-459a-8242-d75e55c1935c.png">

## Testing Instructions

1. Apply D109738-code to your sandbox.
2. Navigate to `/me`.
3. Verify that enabling the 'I am a developer' option and hitting 'Save' sets the `is_dev_account` user attribute.
4. Verify that disabling the option removes the user attribute.
